src/analyzer/analyzersilence.cpp

```diff
@@ -43,34 +41,57 @@ bool AnalyzerSilence::initialize(TrackPointer pTrack,
     }
 
     m_iFramesProcessed = 0;
-    m_bPrevSilence = true;
     m_iSignalStart = -1;
     m_iSignalEnd = -1;
 
     return true;
 }
 
-bool AnalyzerSilence::processSamples(const CSAMPLE* pIn, const int iLen) {
-    for (int i = 0; i < iLen; i += mixxx::kAnalysisChannels) {
-        // Compute max of channels in this sample frame
-        CSAMPLE fMax = CSAMPLE_ZERO;
-        for (SINT ch = 0; ch < mixxx::kAnalysisChannels; ++ch) {
-            CSAMPLE fAbs = fabs(pIn[i + ch]);
-            fMax = math_max(fMax, fAbs);
+// static
+SINT AnalyzerSilence::findFirstSoundInChunk(const CSAMPLE* pIn, SINT iLen) {
+    SINT i = 0;
+    for (; i < iLen; ++i) {
+        if (fabs(pIn[i]) >= kSilenceThreshold) {
+            break;
         }
+    }
+    return i;
```

Comment:
The results of these functions are now inconsistent. findFirstSoundInChunk returns an out-of-range index when not found and findLastSoundInChunk returns the first index.

Reply:
Yes. That's because of going backwards. Is this a problem?

Reply:
No. findLastSoundInChunk returns 0.

Reply:
And -1 if the buffer size was 0.

Reply:
fixed